### PR TITLE
feat: stricter ci

### PR DESCRIPTION
**Pull request by vincentserpoul**
for the golangci-lint, it might be a bit too opinionated, it's probably for the best but in the end, it's your decision, of course

### CODE REVIEW

File: .golangci.yml [hunk 1]
```diff
@@ -1,63 +1,168 @@
...
     - whitespace
-
+    - wrapcheck
+    - wsl
+    - zerologlint
 issues:
-   exclude-rules:
-    - path: _test\.go
+  exclude-rules:
+    - path: (.+)_test\.go
```
Comment (Torakushi) [inline .golangci.yml:148]: Is it possible to remove some linter from the example?

Something like:

```
 - path: cmd/dfns_example/(.+).go
      linters:
        - funlen
        - goconst
        - interfacer
        - dupl
        - lll
        - goerr113
        - varnamelen
        - errcheck
        - gocritic
        - cyclop
        - wrapcheck
        - gocognit
        - contextcheck
        - nlreturn
        - gochecknoglobals
        - errorlint
```

(at least err113, as it makes no sense for an example to make error static)

**Comment by vincentserpoul** [comment]
excluded the cmd folder, it should work.

I think you can also put the folders in pkg directly at the root of the repo.
Usually, the pkg folder is used when you re working on a binary. You can leverage the internal folder though, if you want to make things not accessible to the package user.

One other thing: having a go doc is seen as a sign of good quality.

I have a few packages that I maintain together, you can have a look at [this one](https://github.com/induzo/gocom/tree/main/database/pginit) as an example (not saying it's the best, but i think it gets the job done)

**Comment by vincentserpoul** [comment]
not sure i completely follow what you want to do.
I think the interface should indeed live in the package that s using it or even its own package if more than one package uses it.

on what to expose, the moment it s published, you can t remove any public type/func without changing major version, which can be problematic as it will force you to maintain multi versions. So the less you expose, the better :)

you can do a quick refactor and we ll see if that still fits our usage, it s the best to discuss

**Comment by Torakushi** [comment]
> not sure i completely follow what you want to do. I think the interface should indeed live in the package that s using it or even its own package if more than one package uses it.
> 
> on what to expose, the moment it s published, you can t remove any public type/func without changing major version, which can be problematic as it will force you to maintain multi versions. So the less you expose, the better :)
> 
> you can do a quick refactor and we ll see if that still fits our usage, it s the best to discuss

Thanks! I've created a little PR (Not finished, just to demonstrate what I had in mind. To illustrate/clarify what I said about moving stuff into `internal` folders):
https://github.com/dfns/dfns-sdk-go/pull/6)

**Comment by Torakushi** [comment]
I will merge this PR and fix the linting problems in a separate PR

**Review by Torakushi — APPROVED** [review]
Linter fixed will be in the next PR

**Pull request merged**